src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,704 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Accessible;
+import org.aeonbits.owner.Config;
+import org.aeonbits.owner.ConfigCache;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+/**
+ * Unit test for GATK configuration file handling.
+ *
+ * Created by jonn on 7/19/17.
+ */
+public class ConfigUnitTest extends GATKBaseTest {
+
+    private final static String testResourceDir = publicTestDir + "org/broadinstitute/hellbender/utils/config/";
+
+    // ================================================================================
+    // Helper Methods:
+    // ================================================================================
+
+    private static void listAndStoreConfigToStdOut(final Accessible config) throws IOException {
+
+        config.list(System.out);
+        System.out.println();
+
+        config.store(System.out, "");
+        System.out.println();
+
+        config.storeToXML(System.out, "");
+        System.out.println();
+    }
+
+    private static void validateAndClearProperty(final String key, final Object value) {
+
+        Assert.assertEquals( String.valueOf( System.getProperty(key) ), String.valueOf(value) );
+
+        System.clearProperty(key);
+    }
+
+    // ================================================================================
+    // Tests:
+    // ================================================================================
+
+//    @DataProvider
+//    Object[][] createArgsAndConfigFileOptionsWithExpectedArgs() {
```

Comment:
Why is this `DataProvider` commented out?

Reply:
This data provider is not actually being used for anything.

I'll remove it.